Gemfile

```diff
@@ -10,12 +12,19 @@ group :development do
   gem 'guard-yield', '~> 0.1.0', require: false
   gem 'pry'
   gem 'pry-stack_explorer', platforms: %i[ruby_19 ruby_20 ruby_21]
-  gem 'rubocop', '0.52.1'
+
+  # rubocop:disable Bundler/DuplicatedGem
+  if Hashie::Extensions::RubyVersion.new(RUBY_VERSION) >=
+     Hashie::Extensions::RubyVersion.new('2.4.0')
+    gem 'rubocop', '~> 1.0'
+  else
+    gem 'rubocop', '0.52.1'
+  end
```

Comment:
Should we just drop unsupported rubies and move up?

Reply:
Happy to do that too.  I don't think there's much benefit to such a wide range of Ruby support.  If you specify a minimum you'd like to use, I can update the PR to reflect that.

Reply:
@dblock 2.4 is the floor for this, but if you want to pick 2.5 (which is EOLed for almost a year) or 2.6 (which is EOLing in about 2 months).

Generally I'm a big fan of keeping Ruby version range relatively narrow, and using gem versions to distinguish.

Reply:
I am OK with whatever gives us one single version of Rubocop without the `if`'s.

We tend not to remove rubies in Hashie because it has such incredibly wide use unless there's a good reason, and this is a good reason.

Reply:
Ok.  I've put up a commit that sets the minimum version at 2.4 and updates existing code to reflect that.